src/toc.tsx

```diff
@@ -146,164 +175,35 @@ export class TableOfContents extends Widget {
   }
 
   /**
-   * Rerender after showing.
+   * Callback invoked upon an update request.
+   *
+   * @param msg - message
    */
-  protected onAfterShow(msg: Message): void {
+  protected onUpdateRequest(msg: Message): void {
+    // Don't bother if the ToC is not visible...
+    /* if (!this.isVisible) {
+      return;
+    } */
     this.update();
```

Comment:
~~This is circular. A call to `update` triggers an `onUpdateRequest` call in the next stack frame, so this will essentially animate the ToC, doing a lot of extra work.~~

Edit: nevermind, see above.